src/compiler/binder.ts

```diff
@@ -649,7 +639,7 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
     // unless it is a well known Symbol.
     function getDeclarationName(node: Declaration): __String | undefined {
         if (node.kind === SyntaxKind.ExportAssignment) {
-            return (node as ExportAssignment).isExportEquals ? InternalSymbolName.ExportEquals : InternalSymbolName.Default;
+            return (node).isExportEquals ? InternalSymbolName.ExportEquals : InternalSymbolName.Default;

```

Comment:
Scrolling though the diff but noticed this leftover parens; not sure if it's also leftover anywhere else.

(Formatter when)

Reply:
Yeah, it looks like the eslint `--fix` command left behind a lot of unneeded parenthesis (and some whitespace in other places!), and we have no other lint rule forbidding them.

Reply:
@JoshuaKGoldberg

Reply:
There are more than a few others in this file (L819, L1208, etc.)

Reply:
Yup, [`@typescript-eslint/no-unnecessary-type-assertion`](https://typescript-eslint.io/rules/no-unnecessary-type-assertion) follows in our standard practice of not trying to adjust so much for formatting. https://typescript-eslint.io/linting/troubleshooting/formatting

Looking forward to TypeScript using dprint 😄

Reply:
Personally, I wouldn't call this a formatting concern though. Extra parenthesis are *rarely* something that people do on purpose and, in here, the original code required those parenthesis **not** for formatting purposes. I think it would be reasonable to strip those away while fixing as the original intent is pretty clear here.